packages/compass/src/app/auto-connect.ts

```diff
@@ -3,9 +3,26 @@ import { importConnections } from 'mongodb-data-service';
 import type { ConnectionInfo } from 'mongodb-data-service';
 import { promises as fsPromises } from 'fs';
 import { UUID } from 'bson';
+import { ConnectionString } from 'mongodb-connection-string-url';
+
+function applyUsernameAndPassword(
```

Comment:
What happens if a user passes a `username` or `password` that errors in a connection string? Should we provide some error protection here? I haven't tested locally, but it seems to me like it would fail the `render` inside of the ampersand view which might make things get into an error state a user would be confused by.

Reply:
Apart from the fact that your comment below needs to be addressed, what would an invalid `username` or `password` look like? The only invalid values I can think of are zero-length strings, but then again we’re already not applying those here

Reply:
Can't think of one, probably not a possibility if we're encoding it. 👍